src/libstd/sys/cloudabi/condvar.rs

```diff
@@ -131,9 +136,16 @@ impl Condvar {
                 ..mem::zeroed()
             },
         ];
-        let mut events: [abi::event; 2] = mem::uninitialized();
-        let mut nevents: usize = mem::uninitialized();
-        let ret = abi::poll(subscriptions.as_ptr(), events.as_mut_ptr(), 2, &mut nevents);
+        let mut events: [mem::MaybeUninit<abi::event>; 2] = mem::MaybeUninit::uninit();
+        let mut nevents: mem::MaybeUninit<usize> = mem::MaybeUninit::uninit();
+        let ret = abi::poll(
+            subscriptions.as_ptr(),
+            events.as_mut_ptr(),
```

Comment:
Does this type check? it's going to be passing a `*mut MaybeUninit<event>` to `poll`, right?

Reply:
It does type check, which surprised me at first as well. Looking into the `MaybeUninit` source though, for a type `MaybeUninit<T>`, `as_mut_ptr` returns a `*mut T` instead of a `*mut MaybeUninit<T>` so `poll` does receive a `*mut event`.

Reply:
But `events` is an array, not a `MaybeUninit`, and the `.as_ptr` should be calling the slice method, not the MaybeUninit method.

Reply:
My apologies, you're correct. I'm not sure why the build was successful then. Regardless, since the slice APIs for MaybeUninit are unstable (as discussed in #62732), should I wrap the entire array in a `MaybeUninit`, or is there a better solution?

Reply:
This is the standard library - it can use unstable features.

Reply:
> 'm not sure why the build was successful then. Regardless, since the slice APIs for MaybeUninit are unstable (as discussed in #62732

Are you sure you were building the CloudAbi parts? A plain `./x.py build` won't do that. I honestly don't know how to build-test this code.